File: source/com/barrybecker4/simulation/trading/model/generationstrategy/FlatStrategy.java
/*
 * Copyright by Barry G. Becker, 2015. Licensed under MIT License: http://www.opensource.org/license.MIT
 */

package com.barrybecker4.simulation.trading.model.generationstrategy;

/**
 * @author Barry Becker
 */
public class FlatStrategy extends AbstractGenerationStrategy {

    public String getName() {
        return "flat";
    }

    public String getDescription() {
        return "Does not change at all. Horizontal line.";
    }


    @Override
    public double calcNewPrice(double stockPrice) {
        return stockPrice;
    }

}

File: scala-source/com/barrybecker4/simulation/trading/model/generationstrategy/FlatStrategy.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.trading.model.generationstrategy


/**
  * @author Barry Becker
  */
class FlatStrategy extends IncrementalGenerationStrategy {

  override def name = "Flat"
  override def description = "Does not change at all over time. A horizontal line."
  override def calcNewPrice(stockPrice: Double): Double = stockPrice
}
